### PR TITLE
doc: release notes: Workqueue migration section

### PR DESCRIPTION
**Pull request by trond-snekvik**
Adds a Workqueue API migration section to the nRF Connect SDK 1.6.0
release notes, including a short guide to what changes are required in
the application, as well as links to upstream issues and pull requests.

Signed-off-by: Trond Einar Snekvik <Trond.Einar.Snekvik@nordicsemi.no>

### CODE REVIEW

**Comment by b-gent** [comment]
Added in the Zephyr section of the release notes in https://github.com/nrfconnect/sdk-nrf/pull/4966/files

File: doc/nrf/releases/release-notes-1.6.0.rst [hunk 1]
```diff
@@ -0,0 +1,643 @@
...
+
+For a complete list of |NCS| specific commits, run:
+
+.. code-block:: none
+
+   git log --oneline manifest-rev ^v2.6.0-rc1
+
+The current |NCS| release is based on Zephyr v2.6.0-rc1.
+See the :ref:`zephyr:zephyr_2.6` Release Notes for an overview of the most important changes inherited from upstream Zephyr.
+
```
Comment (jdstapes) [inline doc/nrf/releases/release-notes-1.6.0.rst:476]: Feel free to fix language and references - I suggest that this should be noted here and reference made to the later section.

```suggestion
Important Note: This version of |NCS| has incorporated the use of new Zephyr Workqueue APIs and avoids the use of the deprecated APIs.  For specific information on what has changed and guidance on taking the new APIs into use, see the section "Zephyr Workqueue API Migration" at the bottom of these release notes.
```

File: doc/nrf/releases/release-notes-1.6.0.rst [hunk 2]
```diff
@@ -0,0 +1,643 @@
...
+************
+
+See `known issues for nRF Connect SDK v1.6.0`_ for the list of issues valid for this release.
+
+Zephyr Workqueue API Migration
+******************************
+
+|NCS| 1.6.0 includes changes to the Zephyr Workqueue API introduced as part of `Zephyr pull request #29618 <https://github.com/zephyrproject-rtos/zephyr/pull/29618>`_.
+This pull request deprecates part of the current Workqueue API, and introduces new APIs to cover the same usage scenarios.
+The new API fixes a set of internal issues in the old design, discussed in `Zephyr issue #27356 <https://github.com/zephyrproject-rtos/zephyr/issues/27356>`_.
```
Comment (jdstapes) [inline doc/nrf/releases/release-notes-1.6.0.rst:600]:
```suggestion
The new API fixes issues discussed in `Zephyr issue #27356 <https://github.com/zephyrproject-rtos/zephyr/issues/27356>`_.
```

File: doc/nrf/releases/release-notes-1.6.0.rst [hunk 3]
```diff
@@ -0,0 +1,643 @@
...
+
+It is recommended that all applications migrate to the new ``k_work`` API when upgrading to |NCS| 1.6.0.
```
Comment (jdstapes) [inline doc/nrf/releases/release-notes-1.6.0.rst:602]:
```suggestion
|NCS| code has been migrated for these changes and it is recommended that all applications migrate to the new ``k_work`` API when upgrading to |NCS| 1.6.0.
```

File: doc/nrf/releases/release-notes-1.6.0.rst (hunk 2 above)
Comment (b-gent) [reply doc/nrf/releases/release-notes-1.6.0.rst:600]: done

File: doc/nrf/releases/release-notes-1.6.0.rst (hunk 3 above)
Comment (b-gent) [reply doc/nrf/releases/release-notes-1.6.0.rst:602]: done

File: doc/nrf/releases/release-notes-1.6.0.rst (hunk 1 above)
Comment (b-gent) [reply doc/nrf/releases/release-notes-1.6.0.rst:476]: no need to change this as the migration guide starts right after this line in the original release notes PR.

**Pull request closed without merging**